VoodooPS2Keyboard/VoodooPS2Keyboard.cpp

```diff
@@ -391,12 +391,21 @@ IOACPIPlatformDevice* ApplePS2Keyboard::getBrightnessPanel() {
     };
 
     if (info) {
-        if (info->videoBuiltin != nullptr)
-            panel = getAcpiDevice(getDevicebyAddress(info->videoBuiltin, 0x400));
+        if (info->videoBuiltin != nullptr) {
+            panel = getAcpiDevice(getDevicebyAddress(info->videoBuiltin, kIOACPILCDDisplay));
+
+            //
+            // On some Ice Lake and Comet Lake Laptops, address of display output device
+            // may not export panel information, use 1f for DD1F instead
+            //
+            if (panel == nullptr)
+                if (BaseDeviceInfo::get().cpuGeneration >= CPUInfo::CpuGeneration::IceLake)
+                    panel = getAcpiDevice(getDevicebyAddress(info->videoBuiltin, 0x1f));
```

Comment:
Make this a named macro too? `kIOACPILCDDisplayICL`?

Reply:
Sorry, I should provide more information.
 
Unlike `0x400` and `0x110` with definition, the `0x1f` here is just for default index of this single display output device (`LCD`) when not configured properly on laptops.

On my laptop (UHD620), I have
```
GFX0@20000
- DD01@100
- DD02@2
- DD03@200
- DD04@4
- DD05@5
- DD06@6
- DD07@7
- DD08@8
- DD09@9
- DD0A@a
- DD0B@b
- DD0C@c
- DD0D@d
- DD0E@e
- DD0F@f
- DD1F@400
```
While on my desktop (UHD630), I have
```
GFX0@20000
- DD01@100
- DD02@2
- DD03@300
- DD04@301
- DD05@302
- DD06@6
- DD07@7
- DD08@8
- DD09@9
- DD0A@a
- DD0B@b
- DD0C@c
- DD0D@d
- DD0E@e
- DD0F@f
- DD1F@1f
```
However, on ICL / CML laptops, all of the addresses are not configured by corresponding values, just `0x1` for `DD01` to `0x1f` for `DD1F`.

I suppose that's a minor bug from the reference code. And Dell, with their `BRT6` mechanism, still use static address and naming convention. So they don't need this workaround.
```
GFX0@20000
- CRT0@100
- DP@300
- DP2@301
- DVI@302
- DVI2@303
- DVI3@304
- DP3@305
- LCD@400
```

Reply:
Oh I got that. Maybe just default display without ICL?

Reply:
Maybe just access it by DD1F name in this case?